src/oc/hoc_oop.c

```diff
@@ -657,6 +668,7 @@ void call_ob_proc(Object *ob, Symbol *sym, int narg){
 
    if (ob->template->sym->subtype & CPLUSOBJECT) {
 	hoc_thisobject = ob;
+	gui_redirect_obj_ = ob;
```

Comment:
Should we just use `hoc_thisobject` or is there a scenario where that would not be the same as `gui_redirect_obj_`? What are the memory management implications? (i.e. when do we ref and unref ob?)

Reply:
I need to move the added line up one line as it obscured the reason for ```hoc_thisobject = ob```/.  It's purpose is to get it into the ```push_frame(sym, narg);``` and the actual value of
hoc_thisobject is obsav when the macro is called. If the macro could look into the top of
the frame list, that would be sufficient. However the frame list is local to src/oc/code.c.
So... could have implemented nrn_get_gui_redirect_obj in code.c without using up an
extra 8 bytes for ```static Object* gui_redirect_obj_``` but then it the semantic information
would be in three places: the macros, the top frame, and call_ob_proc.

Reply:
Looks like ob is very often not hoc_thisobject in call_ob_proc. I put an assert statement
near the beginning of that  function to check and
```
oc>objref x
oc>x = new Vector(5)
oc>x.size()
Assertion failed: file /home/hines/neuron/nrncmake/src/nrnoc/../oc/hoc_oop.c, line 653
nrniv: ob == hoc_thisobject
```